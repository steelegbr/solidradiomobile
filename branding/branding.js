/**
 * Provides branding functions.
 */

import { DefaultTheme, DarkTheme } from 'react-native-paper';

/**
 * Generates a station theme.
 * @param {bool} darkMode An indication if we are in dark mode.
 * @param {station} station The station this theme is for.
 */

export function generateStationTheme(darkMode, station) {

    let theme = generateTheme(darkMode);
    theme.colors.primary = station.primary_colour;
    theme.colors.accent = station.primary_colour;
    theme.mode = 'adaptive';
    return theme;

}

/**
 * Generates a theme with no station.
 * @param {bool} darkMode An indication if we are in dark mode.
 */

export function generateTheme(darkMode) {

    if (darkMode) {
        return {
            ...DarkTheme,
            roundness: 10,
            colors: {
                ...DarkTheme.colors
            }
        };
    } else {
        return {
            ...DefaultTheme,
            roundness: 10,
            colors: {
                ...DefaultTheme.colors
            }
        };
    }

}
