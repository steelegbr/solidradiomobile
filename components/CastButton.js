/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import React, { Component } from 'react';
import { Button } from 'react-native-paper';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/MaterialIcons';

class CastButton extends Component {

    render() {

        const { icon, style, bigMode, iconSize } = this.props;

        if (bigMode) {
            return (
                <Icon.Button style={style} name={icon} color={style.color} backgroundColor={style.backgroundColor} size={iconSize} />
            );
        } else {
            return (
                <Icon.Button style={style} name={icon} color={style.color} backgroundColor={style.backgroundColor} size={16} />
            );
        }

    }

}

function mapStateToProps(state, ownProps) {

    const bigMode = 'bigMode' in ownProps ? ownProps.bigMode : false;
    const iconSize = 'iconSize' in ownProps ? ownProps.iconSize : 0;

    return {
        icon: "cast",
        style: {
            color: state.theme.colors.primary,
            backgroundColor: 'rgba(0, 0, 0, 0.0)',
            padding: 10,
            margin: 5
        },
        bigMode: bigMode,
        iconSize: iconSize
    };

}

export default connect(mapStateToProps)(CastButton);