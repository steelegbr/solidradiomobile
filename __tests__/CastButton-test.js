/**
 * Tests the Chromecast button.
 */

import React from 'react';
import CastButton from '../components/CastButton';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { generateTheme } from '../branding/branding';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);
let store;

beforeEach(() => {
    store = mockStore({
        theme: generateTheme(false)
    });
});

it('renders-default', () => {

    const castButton = renderer.create(
        <Provider store={store}>
            <CastButton />
        </Provider>
    ).toJSON();
    expect(castButton).toMatchSnapshot();

});

it('renders-bigmode', () => {

    const castButton = renderer.create(
        <Provider store={store}>
            <CastButton bigMode={true} iconSize={32} />
        </Provider>
    ).toJSON();
    expect(castButton).toMatchSnapshot();

});