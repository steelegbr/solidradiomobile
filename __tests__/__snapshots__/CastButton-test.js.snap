// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders-bigmode 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "#f6f6f6",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 32,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders-default 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "#f6f6f6",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 16,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
